codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyrust/generator/AbstractRustShimGenerator.java

```diff
@@ -463,7 +463,10 @@ protected TokenTree fromDafny(
               valueToRust = "(%s).unwrap()".formatted(valueToRust);
             }
           } else {
-            valueToRust = dafnyToRust.formatted(dafnyValue + ".as_ref()");
+            valueToRust =
+              dafnyToRust.formatted(
+                "::std::borrow::Borrow::borrow(%s)".formatted(dafnyValue)
+              );
```

Comment:
For my own education, what's the difference?

Reply:
https://doc.rust-lang.org/std/convert/trait.AsRef.html#relation-to-borrow

The relevant point here is that `Borrow` has a blanket impl for any `T` (whereas both `Borrow` and `AsRef` have blanket impls for any `&T` and `&mut T`).